website/source/docs/connect/proxies/envoy.md

```diff
@@ -14,257 +14,309 @@ optionally exposing a gRPC service on the local agent that serves [Envoy's xDS
+  metrics endpoint to the _public_ network. It must be supplied in the form
+  `ip:port` and port and the ip/port combination must be free within the network
+  namespace the proxy runs. Typically the IP would be `0.0.0.0` to bind to all
+  available interfaces or a pod IP address.
+
+    -> **Note:** Envoy versions prior to 1.10 do not export timing histograms
+    using the internal prometheus endpoint. Consul 1.5.0 [doesn't yet support
+    Envoy 1.10](#supported-versions) although support will soon be added.
+
+- `envoy_stats_tags` - Specifies one or more static tags that will be added to
+  all metrics produced by the proxy.
+
+- `envoy_stats_flush_interval` - Configures Envoy's
+  [`stats_flush_interval`](https://www.envoyproxy.io/docs/envoy/v1.9.1/api-v2/config/bootstrap/v2/bootstrap.proto#envoy-api-field-config-bootstrap-v2-bootstrap-stats-flush-interval).
+
+There are more possibilities available in the [Advanced
+Configuration](#advanced-configuration) section that allow incremental or
+complete control over the bootstrap configuration generated.
+
+## Dynamic Configuration
+
+Envoy's dynamic configuration is generated by Consul and is automatic. For Envoy
+we generate different listener configurations depending on the service's
+protocol which is defined using [`service-defaults` configuration
+entries](#TODO).
+
+The following configuration items may be added to the [global `proxy-defaults`
+configuration entry](#TODO) or overridden directly in the `proxy.config` field
+of a [proxy service
+definition](/docs/connect/proxies.html#proxy-service-definitions) or
+[`sidecar_service`](/docs/connect/proxies/sidecar-service.html) block.
+
+- `protocol` - The protocol the service speaks. The following `protocol` values
+  are supported by Connect's Envoy integration currently:
+  - `tcp` - The default is plain TCP and will proxy at L4. This gets all the
+    security benefits of Connect's mTLS and works for any TCP-based protocol.
+    Load-balancing and metrics are available at the connection level.
+  - `http` - This specified the service speaks HTTP/1.x. Envoy will setup an
+    `http_connection_manager` and will be able to load-balance requests
+    individually to available upstream services. Envoy will also emit L7 metrics
+    such as request rates broken down by HTTP response code family (2xx, 4xx, 5xx,
+    etc).
+  - `http2` - This specifies that the service speaks http2. Specifically h2c since
+    Envoy will still only connect to the local service instance via plain TCP not
+    TLS. This behaves much like `http` with L7 load-balancing and metrics but has
+    additional settings that correctly enable end-to-end http2. Envoy in the
+    future may support automatic upgrades between HTTP/1.x and http2 and not need
+    explicit configuration.
+  - `grpc` - gRPC is a common RPC protocol based on http2. In addition to the
+    http2 support above, a service with `grpc` protocol will be configured with a
```

Comment:
```suggestion
    http2 support above, Envoy will configure `grpc` protocol services with a
```

Reply:
Give this a good technical check, I may not have captured the meaning when changing it from passive voice

Reply:
```suggestion
    http2 support above, Envoy listeners will be configured with a
```